packages/stockflux-launcher/src/App.js

```diff
@@ -1,66 +1,144 @@
-import React, { useRef, useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 import cx from 'classnames';
-import { OpenfinApiHelpers } from 'stockflux-core';
-import { useDockWindow, ScreenEdge } from 'openfin-react-hooks';
+import { useDockWindow, useOptions, ScreenEdge } from 'openfin-react-hooks';
 import {
   FaChevronUp,
   FaChevronLeft,
   FaChevronRight,
-  FaRegHandRock
+  FaRegHandRock,
+  FaUnlock
 } from 'react-icons/fa';
-import AppShortcuts from './app-shortcuts/AppShortcuts';
 import FreeTextSearch from './free-text-search/FreeTextSearch';
 import ToolBar from './toolbar/ToolBar';
 import CloseButton from './toolbar/CloseButton';
 import './App.css';
 
-const isLauncherHorizontal = edge => edge === ScreenEdge.TOP;
-
 export default () => {
+  const [alignment, setAlignment] = useState('horizontal');
+  const [isCloudMode, setCloudMode] = useState(false);
+  const [isDocked, setDocked] = useState(true);
+  const [options] = useOptions();
+  const [undockWidth, setUndockWidth] = useState(1000);
+  const [undockHeight, setUndockHeight] = useState(50);
+  const [undockTop] = useState(50);
+  const [undockLeft, setUndockLeft] = useState(0);
+
   const [edge, windowActions] = useDockWindow(
     ScreenEdge.TOP,
-    OpenfinApiHelpers.getCurrentWindowSync(),
+    window.fin.Window.getCurrentSync(),
     true,
-    { dockedWidth: 50, dockedHeight: 50 }
+    { dockedWidth: 50, dockedHeight: 50 },
+    {
+      undockPosition: { top: undockTop, left: undockLeft },
+      undockSize: { width: undockWidth, height: undockHeight }
+    }
   );
 
-  const prevEdgeRef = useRef();
+  let currentDirection = edge !== ScreenEdge.NONE ? edge : ScreenEdge.TOP;
+  let left =
+    window.screenLeft >= window.screen.availWidth ? window.screenLeft : 0;
+  let right =
+    window.screenLeft >= window.screen.availWidth
+      ? window.screenLeft
+      : window.screen.availWidth - 50;
+  let midpoint =
+    left === 0
+      ? (left + right) / 4
+      : window.screen.availWidth + (left + right) / 8;
```

Comment:
Spoke with @kangelSL and thinking about pulling a lot of the screen positioning maths out into separate functions. So a additional issue will be raised later to add this into said functions.

Reply:
Clarified position of undocked; undocked will always be in horizontal position which made simplifying code easier. Moved position code to its own function